### ISSUE TITLE
Aur block idea

### ISSUE THREAD

**Issue by Stoope** [greshake/i3status-rust#385]
I have an idea of create a block which displays updates count from AUR. Like pacman block. It'll have source parameter (yay, trizen, pikaur, ...).
What do you think? Or may be it is better to extend existing pacman block? I can try to implement it.

**Comment by atheriel** [comment]
Not strongly opposed to this, but we have avoided including AUR updates in the past because of churn in AUR tools.

It's pretty easy to make a Custom block for this, if you need it now.

**Comment by rsteube** [comment]
Needed AUR updates as well and adjusted the pacman block for [pamac](https://wiki.manjaro.org/index.php/Pamac) which is the default package manager on Manjaro.

https://github.com/rsteube/i3status-rust/commit/876f3f66bf58217632ac6939b63a6b0f17f9b848

**Comment by ammgws** [comment]
Perhaps the pacman block could be extended with support for checking the output of a command supplied in the i3status-rust's config file.

**Comment by ammgws** [comment]
@PicoJr Not a release blocker, but perhaps you might like to work on this? I was thinking something like:

- add an optional `aur_command` to the config, and the pacman block can run that to determine how many AUR updates there are. Since there are many AUR helpers, it's up to the user to supply the appropriate command.

- add a formatter to the pacman block with qualifiers {pacman}, {aur}, {both} so users can format the block however they want

- document that the update interval should be set appropriately as to not exceed the AUR's daily rate limit

**Comment by PicoJr** [comment]
> Add an optional `aur_command` to the config

Ok, the supplied command should output available updates to stdout. Then we can process the output the same way we do with pacman.

I guess it should work: I don't use an AUR helper for now ^^'

> Add a formatter to the pacman block with qualifiers `{pacman}`, `{aur}`, `{both}`

I am not sure I understand what it means.

Does it mean:

* `{pacman}` = update count for Arch Repository (AR)
* `{aur}` = update count for Arch User Repository (AUR)
* `{both}` = update count for AR + update count for AUR

Do we keep `{count}` or do we replace it with `{pacman}`? (breaking change)

```toml
# config for a pacman user
format = "{pacman} updates available"
format_singular = "{pacman} update available"
format_up_to_date = "system up to date"
```

```toml
# config for an AUR helper user
format = "{aur} updates available"
format_singular = "{aur} update available"
format_up_to_date = "system up to date"
```

```toml
# config for a pacman + AUR helper user who wants to aggregate both
format = "{both} updates available"
format_singular = "{both} update available"
format_up_to_date = "system up to date"
```

```toml
# config for a pacman + AUR helper user who wants both AR and AUR update counts displayed
format = "{pacman} and {aur} (AUR) updates available"
# format_singular does not really makes sense here
format_singular = "{pacman} and {aur} (AUR) update available"
format_up_to_date = "system up to date"
```

**Comment by ammgws** [comment]
> I am not sure I understand what it means.

Ah, I was going off the top of my head. Could change `pacman` to `count`, or just have both for now and deprecate count in the future.

**Comment by ammgws** [comment]
Implemented via #385 thanks to @PicoJr
